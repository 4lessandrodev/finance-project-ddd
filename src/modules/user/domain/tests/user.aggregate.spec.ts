import UserMock from './mock/user.mock';

describe('user.aggregate', () => {

	const userMock = new UserMock();

	it('should create a valid user', () => {
		const user = userMock.domain();

		expect(user.isSuccess).toBe(true);
	});

	it('should get valid values', () => {
		const user = userMock.domain();

		const userResult = user.getResult();

		expect(userResult.id).toBeDefined();
		expect(userResult.createdAt).toBeDefined();
		expect(userResult.email.value).toBe('valid_email@domain.com');
		expect(userResult.isDeleted).toBeFalsy();
		expect(userResult.password.value).toBe('valid_password');
		expect(userResult.terms).toMatchSnapshot();
	});

	it('should create a valid user with provided id', () => {
		const user = userMock.domain({ id: 'valid_id' });

		expect(user.getResult().id.toValue()).toBe('valid_id');
	});
});
