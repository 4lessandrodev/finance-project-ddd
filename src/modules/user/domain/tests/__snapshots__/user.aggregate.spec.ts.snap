// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user.aggregate should get valid values 1`] = `
Array [
  TermValueObject {
    "props": Object {
      "acceptedAt": DateValueObject {
        "ONE_DAY": 86400000,
        "ONE_HOUR": 3600000,
        "ONE_MINUTE": 60000,
        "ONE_MONTH": 2678400000,
        "ONE_WEEK": 604800000,
        "ONE_YEAR": 31622400000,
        "props": Object {
          "value": 2022-01-01T00:00:00.000Z,
        },
      },
      "ip": IpValueObject {
        "props": Object {
          "value": "127.0.0.1",
        },
      },
      "isAccepted": true,
      "userAgent": Object {
        "name": "firefox",
        "os": "linux",
        "type": "browser",
        "version": "86.0.1",
      },
    },
  },
]
`;
